### PR TITLE
Add helperText prop to ArrayInput

### LINKED ISSUE #4227: ArrayInput does not support helperText prop
**What you were expecting:**
As ArrayInput should accept the common input props, helperText should be working, too.

**What happened instead:**
HelperText prop is not handled in the input, therefore does not appear.

**Related code:**
I'd love to create a pull request, however, I am not sure how the related files are compiled. Making the prop work would only need a few lines of code into ArrayInput.tsx:

```
import {  InputLabel, FormControl, FormHelperText } from '@material-ui/core';
...
<InputLabel ... />
<FormHelperText>{helperText}</FormHelperText>
```
Besides this, adding helperText const to the props and the proptypes.

**Environment**

* React-admin version: 3.1.1

### PR DESCRIPTION
**Pull request by netdown**
Fixes #4227 

I was not able test - I am not sure of building -, sorry about that, but as it is only a 5 line change, there should be no problem.

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/ArrayInput.tsx [hunk 1]
```diff
@@ -85,6 +86,8 @@ export const ArrayInput = ({
                     isRequired={isRequired(validate)}
                 />
             </InputLabel>
+            {helperText &&
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/ArrayInput.tsx]: We might have validation applied to the `ArrayInput` itself, in which case the `HelperText` should display the error instead of the helper text like the other inputs do. You can find a very close implementation in `SelectArrayInput`.

Comment (netdown) [reply packages/ra-ui-materialui/src/input/ArrayInput.tsx]: Updated, however I am not sure of the props useInput requires, do all inputs receive them?

**Comment by djhi** [comment]
Looks good. However you'll have to pass the `error` to the `InputLabel` too

File: packages/ra-ui-materialui/src/input/ArrayInput.tsx [hunk 2]
```diff
@@ -85,6 +109,15 @@ export const ArrayInput = ({
                     isRequired={isRequired(validate)}
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/ArrayInput.tsx]: `isRequired` is already a boolean

File: packages/ra-ui-materialui/src/input/ArrayInput.tsx [hunk 3]
```diff
@@ -69,6 +76,23 @@ export const ArrayInput = ({
         validate: sanitizedValidate,
         ...rest,
     });
+    
+    const {
+        id,
+        input,
+        isRequired,
+        meta: { error, touched },
+    } = useInput({
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/ArrayInput.tsx]: You don't need `useInput`, `useFieldArray` returns a `meta` which contains the `error` if any. See https://github.com/final-form/react-final-form-arrays#fieldarrayrenderprops

File: packages/ra-ui-materialui/src/input/ArrayInput.tsx (hunk 2 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/ArrayInput.tsx]: Ignore previous comment. As you'll not be using `useInput`, you'll have to call `isRequired(validate)`. Just not the `isRequired` from the `useInput` but the function imported

**Comment by fzaninotto** [comment]
As it's a new feature, please PR against `next` rather than `master`

**Comment by netdown** [comment]
> Looks good. However you'll have to pass the `error` to the `InputLabel` too for it to become red on error

Neither component passes that prop.

**Comment by djhi** [comment]
> Neither component passes that prop

See https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/input/SelectArrayInput.tsx#L215

PS: This was merged a few hours ago (#4249)

**Comment by netdown** [comment]
@fzaninotto Are you sure it should be marked as a new feature? The documentation suggests that this prop is also available on ArrayInput, therefore I'd consider it as a fix.

**Comment by fzaninotto** [comment]
The current documentation in wrong, then ;). Yes, it's a new feature - it was never implemented.

**Comment by netdown** [comment]
@djhi Any suggestions on CI failure? The error prop is of type any in useInput and final form: https://final-form.org/docs/react-final-form/types/FieldRenderProps#metaerror

**Comment by djhi** [comment]
First, can you please rebase your branch on `next`. The diff is currently very hard to follow

**Comment by fzaninotto** [comment]
Superseded by #4262
